### DISCUSSION TITLE
Any thoughts of creating a Tdarr plugin to utilize this?

### DISCUSSION THREAD

**Discussion by feltz916** [absadiki/subsai#52] [Ideas]
Great work btw!

Wish I knew JavaScript, but id love to find a way to automate this somehow.

**Comment by absadiki** [comment]
Thanks @feltz916,

What is Tdarr, you want to automate what ? could you please explain more ?

**Reply by feltz916** [reply]
Hi @abdeladim-s , Tdarr is a media library distributed transcoding software. Most people use it to process videos automatically for various functions (like transcoding into standardized formats, health checks, etc). I personally use it to process all downloaded videos and compress them into HEVC codec to save space.

I would like to be able to automate the process of subtitle generation during my re-encoding phase, by calling upon your tool.

Then the media has an associated subtitle SRT file with it. (Currently, I do this through Plex, which is pretty horrible as it tries to download subtitles through 3rd party services, which not only supplies ads with it but also quite often is not in sync, so it useless.

I will probably attempt to hook it up myself although my programming knowledge does not cover JavaScript (which is what Tdarr plugins use).

**Reply by absadiki** [reply]
Yeah I see .. good idea.

I don't know much about Tdarr, but from what I read from their docs, the idea I have in mind is you can create a [post-processing plugin](https://github.com/HaveAGitGat/Tdarr_Plugins/blob/master/examples/Tdarr_Plugin_pos1_Post_Proc_Example.js) on your re-encoding phase, and from inside the plugin function you can use a [child process](https://nodejs.org/api/child_process.html) to call `subsai` command line interface to generate the subtitle. 

But I don't know how doable this is, or if Tdarr plugin worker allows spawning child processes!

Anyways, give it a try and let us know if you succeeded or if you need any help during the process.  

Good luck!
